socket/src/commonTest/kotlin/mqtt/socket/SocketTests-2.kt

```diff
@@ -0,0 +1,178 @@
+    @ExperimentalTime
+    @Test
+    fun oneServerOneClient() = block {
+        var port : UShort = 0u
+        lateinit var server : TCPServer
+        lateinit var client: ClientToServerSocket
+
+        val serverProcess = ServerProcessTest(ServerAction.MQTTSTRING)
+        serverProcess.name = "Server-1"
+        serverProcess.clientResponse = "Client-"
+        server = TCPServer ("localhost", port, serverProcess)
+        launchServer (this, port, server)
+
+        port = server.getListenPort()
+
+        client = asyncClientSocket()
+        initiateClient(client, port)
+
+        clientMessage (client, "Client-1", "Client-1:Server-1")
+        client.close()
+        server.close()
+    }
+
+    @ExperimentalUnsignedTypes
+    @ExperimentalTime
+    @Test
+    fun oneServerMultiClient() = block {
+        var port: UShort = 0u
+        val clientCount = 6
+        val serverProcess = ServerProcessTest(ServerAction.MQTTSTRING)
+        serverProcess.name = "Server-x"
+        serverProcess.clientResponse = "Client-"
+        val server = TCPServer("localhost", port, serverProcess)
+        launchServer(this, port, server)
+        port = server.getListenPort()
+
+        val doneMutex = Mutex(true)
+        closeCounter.counter = 0
+        repeat(clientCount) { i ->
+            launch (Dispatchers.Default) {
+                clientSetup(port, port, i, clientCount, doneMutex)
+            }
+        }
+        doneMutex.lock()
+        server.close()
+        assertEquals(0, readStats(port, "CLOSE_WAIT").count(), "sockets found in close_wait state")
+        doneMutex.unlock()
+    }
+
+    @ExperimentalUnsignedTypes
+    @ExperimentalTime
+    @Test
+    fun twoServerMultiClient() = block {
+        var port0: UShort = 0u
+        var port1: UShort = 0u
+        val clientCount = 8
+        val serverProcess = ServerProcessTest(ServerAction.MQTTSTRING)
+        serverProcess.name = "Server-x"
+        serverProcess.clientResponse = "Client-"
+
+        val server0 = TCPServer("localhost", port0, serverProcess)
+        val server1 = TCPServer("localhost", port1, serverProcess)
+        launchServer(this, port0, server0)
+        launchServer(this, port1, server1)
+        port0 = server0.getListenPort()
+        port1 = server1.getListenPort()
+
+        val doneMutex = Mutex(true)
+        closeCounter.counter = 0
+        repeat(clientCount) { i ->
+            launch (Dispatchers.Default){
+                clientSetup(port0, port1, i, clientCount, doneMutex)
+            }
+        }
+        doneMutex.lock()
+        server0.close()
+        server1.close()
+        assertEquals(0, readStats(port0, "CLOSE_WAIT").count(), "sockets found in close_wait state")
+        assertEquals(0, readStats(port1, "CLOSE_WAIT").count(), "sockets found in close_wait state")
+        doneMutex.unlock()
+    }
+
+    companion object closeCounter {
+        var counter = 0
+        private val mux: Mutex = Mutex()
+        suspend fun increment() {
```

Comment:
is AtomicInt not available?

https://kotlinlang.org/docs/reference/coroutines/shared-mutable-state-and-concurrency.html
might be of help

It just seems weird that we are locking and unlocking just to increment

Reply:
I was not aware of AtomicInt. I will change to use AtomicInt.